substrate/frame/revive/src/storage/meter.rs

```diff
@@ -382,12 +355,18 @@ where
 				Origin::Root => return Ok(Deposit::Charge(Zero::zero())),
 				Origin::Signed(o) => o,
 			};
-			for charge in self.charges.iter().filter(|c| matches!(c.amount, Deposit::Refund(_))) {
-				E::charge(origin, &charge.contract, &charge.amount, &charge.state)?;
-			}
-			for charge in self.charges.iter().filter(|c| matches!(c.amount, Deposit::Charge(_))) {
-				E::charge(origin, &charge.contract, &charge.amount, &charge.state)?;
-			}
+			let try_charge = || {
+				for charge in self.charges.iter().filter(|c| matches!(c.amount, Deposit::Refund(_)))
+				{
+					E::charge(origin, &charge.contract, &charge.amount, &charge.state)?;
+				}
```

Comment:
can these refunds ever fails :thinking: ?

Reply:
Yes of course. And this is also exercised in the tests: You can just choose a limit higher than the balance of the origin. Or the origin can send balance away by calling into a pre-compile.